Update kedro pipeline create to use new /conf file structure

DimedS opened the pull request:
Fixes #2752

## Description
This Pull Request (PR) refines the Kedro pipeline create command. Instead of nesting a new "{pipeline-name}.yml" file in a "/parameters" folder, we now directly create a "parameters_{pipeline-name}.yml" file in the conf folder. This change simplifies the directory structure and improves user accessibility.

## Development notes
I moved the {pipeline-name}.yml /template file from the parameters folder to the conf folder and added "parameters_" at the start of the filename.

This change meant I had to tweak the delete_pipeline function in cli/pipeline.py so it knew where to find the right files to delete.

After these changes, a lot of tests stopped working because they were set up for the old folders. I fixed these tests in test_pipeline.py.

I also had to make changes for the microPackaging. I added two new search templates to 'configs_to_package' in cli/micropkg.py, but I kept the two old ones that use the "parameters" folder, so we don't break anything for users who use the old structure.

I fixed the tests for these changes in test_micropkg_package.py and test_micropkg_pull.py.

While fixing the tests, I noticed I was repeating a lot of the same fixes. We should look at redesigning the tests later to make things more consistent.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

merelcht reviewed (commented):
Congrats on your first PR @DimedS 😄 ⭐ 

The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

noklam reviewed (commented):
Leave some comments. But more importantly I notice `micropkg` is completely broken yet our `e2e-test` have no problem about it. I am not an expert on `micropkg` so I need some help.

See the directory here when I do `kedro micropkg pacakge pipelines.nok` and `tar -xf` in `main` branch

```
├── MANIFEST.in
├── nok
│   ├── config
│   │   └── parameters_nok.yml
│   ├── __init__.py
│   ├── nodes.py
│   ├── pipeline.py
│   └── README.md
├── nok.egg-info
│   ├── dependency_links.txt
│   ├── PKG-INFO
│   ├── SOURCES.txt
│   └── top_level.txt
├── PKG-INFO
├── pyproject.toml
├── setup.cfg
└── tests
    ├── __init__.py
    └── test_pipeline.py
```

I expect this should be `conf` instead of `config`, but it seems that this has been like this for long time, do we expect a micropkg package is `kedro runable`?. @merelcht

noklam commented:
> Congrats on your first PR @DimedS 😄 ⭐
> 
> The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

Should this be a e2e-test?

DimedS commented:
> Congrats on your first PR @DimedS 😄 ⭐
> 
> The code looks good to me! The only question I have is around the backwards compatibility for `kedro pipeline delete`, if you run that command with this new code but on an "old" pipeline e.g. one with a `parameters/pipeline.yml` structure it won't delete those parameter files properly. Is it possible to make it work with both parameters inside a directory and outside?

Merel, thank you for your insights, I fully agree! I've added the requested feature, along with the corresponding unit tests. Additionally, I've updated the Release.MD file and re-requested a review.

DimedS commented on kedro/framework/cli/micropkg.py:
```diff
@@ -613,9 +613,16 @@ def _package_micropkg(
     configs_to_package = _find_config_files(
         package_conf,
-        [f"parameters*/**/{micropkg_name}.yml", f"parameters*/**/{micropkg_name}/**/*"],
+        [
+            f"**/parameters_{micropkg_name}.yml",
+            f"**/{micropkg_name}/**/*",
+            f"parameters*/**/{micropkg_name}.yml",
+            f"parameters*/**/{micropkg_name}/**/*",
+        ],
```
Nok hi, thank you for your review. In my opinion, currently it's not configurable. However, as you've suggested, it seems like a good idea to refactor the micropkg module code in a separate ticket. At this time, I've simply updated the default 'parameters' structure while maintaining backward compatibility.

noklam commented on kedro/framework/cli/pipeline.py:
```diff
@@ -140,10 +140,19 @@ def delete_pipeline(
         for confdir in ("parameters", "catalog")
         if (pipeline_artifacts.pipeline_conf / confdir / f"{name}.yml").is_file()
     ]
+
```
Can it be refactored as looping the two patterns? There are many repeat code here, the only difference is `f"{confdir}_{name}.yml"` and  `confdir / f"{name}.yml"`

DimedS replied:
Thanks for catching that, Nok. Fixed it  👍

merelcht reviewed (approved):
Thanks @DimedS, great work! ⭐

noklam reviewed (approved):
Thanks for addressing the comment. I think this is good enough to be merged. What's your thought about the e2e test, do you still think it is worth it?

noklam commented:
I think this is ready and I have approved it already.
